doc/flow.rst

```diff
@@ -0,0 +1,75 @@
+*********
+Code Flow
+*********
+
+Overview
+========
+
+Setup
+=====
+
+Flow
+----
+
+Below is the code flow for the :ref:`pypit-setup` script.  The
+following are nearly all function names or object methods.
+The module name is typically the first item, e.g. arparse.init
+is a method in arparse.py.  Here goes::
+
+   ├── pypit_setup
+   |  ├── pyputils.make_pypit_file(pypit_file, spectrograph, dfnames)
+   |  ├── run_pypit.parser(options)
+   |  ├── pypit.PYPIT(args)
+   |  |  ├── load_input(pypit_file)
+   |  |  |  ++ generates pyp_dict
+   |  |  ├── arparse.get_argflag_class()
+   |  |  |  ++ generates argf Class
+   |  |  ├── argf.init_param()
+   |  |  ├── plines = argf.load_lines(parlines)
+   |  |  ├── argf.set_paramlist(plines)
+   |  |  ├── arparse.get_spect_class()
+   |  |  |  ++ generates spect Class
+   |  |  ├── spect.load_file(base=True)  # default
+   |  |  ├── spect.set_paramlist(lines)
+   |  |  ├── spect.load_file()  # instrument specific
+   |  |  ├── spect.set_paramlist(lines)
+   |  |  ├── spect.set_paramlist(plines)   # using pyp_dict
+   |  |  ├── spect.load_lines(spclines)      # command line
+   |  |  ├── argf.save()
+   |  |  ├── spect.save()
+   |  |  ├── arparse.init(argf, spect)  # Saved into settings.
+   |  |  ├── # fitsdict created
+   |  |  ├── fitsdict = arload.load_headers(datlines)
+   |  |  |  ├── # Checks on FITS files
+   |  |  |  ├── settings.spect['check'][ch]
+   |  |  ├── # Flip dispersion direction (if needed)
+   |  ├── ARMLSD  # This is formally below PYPIT, but I want to reduce the indentation here
+   |  |  ├── armbase.SetupScience(fitsdict)
+   |  |  |  ├── filesort = arsort.sort_data(fitsdict)
+   |  |  |  |  ├── find_standard_file()
+   |  |  |  |  ++ Generates filesort dict
+   |  |  |  ├── arsort.match_science(fitsdict, filesort)
+   |  |  |  |  ++ Written to settings.spect[ftag]['index']
+   |  |  |  ├── arsciexp.ScienceExposure(i, fitsdict)
+   |  |  |  |  ++ Generates sciexp list of ScienceExposure objects
+   |  |  |  ++ setup_dict generated
+   |  |  |  ├── arsort.instr_setup(sciexp, kk+1, fitsdict, setup_dict)
+   |  |  |  |  ++ Generates setupIDs
+   |  |  |  ++ group_dict generated
+   |  |  |  ├── arsort.write_sorted(group_dict, setup_dict)
+   |  |  |  ├── arsort.write_setup(setup_dict)
+
+
+Items
+-----
+
+Items created and carried around::
```

Comment:
Would it be useful if we had a short description of what information each of these items carries around? Might not be of interest for people using PYPIT just for data reduction, but I think it helps clarify the flow a bit more.

For example, 
settings.argf: parameters for the data reduction, such as the method for combining calibration frames; includes user-specified parameters to overwrite the default from the user-input .pypit file

I see now as I'm trying to describe these items that they're not so easily describe-able. Some of them contain a lot of information...I'm not sure if we plan to have documentation on all of the argflag parameters that the user can tweak, but if we do in the future (I'm happy to help), we can link to it here.

Anyway, everything else looks good to me!

Reply:
Excellent comment.  I've started doing something quite 
like this in another PR so am going to defer it for now.